CorsixTH/Lua/app.lua

```diff
@@ -1680,60 +1680,136 @@ function App:loadLuaFolder(dir, no_results, append_to)
-  elseif ver > 170 then
-    return "v0.67"
-  elseif ver > 156 then
-    return "v0.66"
-  elseif ver > 138 then
-    return "v0.65"
-  elseif ver > 134 then
-    return "v0.64"
-  elseif ver > 127 then
-    return "v0.63"
-  elseif ver > 122 then
-    return "v0.62"
-  elseif ver > 111 then
-    return "v0.61"
-  elseif ver > 105 then
-    return "v0.60"
-  elseif ver > 91 then
-    return "0.50"
-  elseif ver > 78 then
-    return "0.40"
-  elseif ver > 72 then
-    return "0.30"
-  elseif ver > 66 then
-    return "0.21"
-  elseif ver > 54 then
-    return "0.20"
-  elseif ver > 53 then
-    return "0.11"
-  elseif ver > 51 then
-    return "0.10"
-  elseif ver > 45 then
-    return "0.01"
-  else
-    return "Beta 8 or earlier"
+--[[
+  CorsixTH versioning follows a convention similar to semantic versioning but it
+  does not accurately follow its standard. As we are currently <1.0.0 the major
+  integer is not at play. Instead:
+  Minor - denotes an new milestone of the program that often encompassses multiple
+  patches that could break savegames without in-situ afterLoads
+  Revision - denotes a revised hotfix release of the program that patches a major
+  bug before the next milestone. The bug is then patched for the next milestone.
+  Moving forward, revisions should have a savegame_version bump of 1, as 10
+  savegame versions are reserved each release for patching.
+  All beta versions must have a savegame increment each time
+  Each patch note must begin with a '-'
+--]]
+local release_table = {
+  -- Format: major, minor, revision, patch (string), savegame_version
+  {major = 0, minor = 0, revision = 8, patch = "", version = 0}, -- Beta 8 or below
+  {major = 0, minor = 1, revision = 0, patch = "", version = 51},
+  {major = 0, minor = 10, revision = 0, patch = "", version = 53},
+  {major = 0, minor = 11, revision = 0, patch = "", version = 54},
+  {major = 0, minor = 20, revision = 0, patch = "", version = 66},
+  {major = 0, minor = 21, revision = 0, patch = "", version = 72},
+  {major = 0, minor = 30, revision = 0, patch = "", version = 78},
+  {major = 0, minor = 40, revision = 0, patch = "", version = 91},
+  {major = 0, minor = 50, revision = 0, patch = "", version = 105},
+  {major = 0, minor = 60, revision = 0, patch = "", version = 111},
+  {major = 0, minor = 61, revision = 0, patch = "", version = 122},
+  {major = 0, minor = 62, revision = 0, patch = "", version = 127},
+  {major = 0, minor = 63, revision = 0, patch = "", version = 134},
+  {major = 0, minor = 64, revision = 0, patch = "", version = 138},
+  {major = 0, minor = 65, revision = 0, patch = "", version = 156},
+  -- There was also 0.65.1, not differentiated by version number
+  {major = 0, minor = 66, revision = 0, patch = "", version = 170},
+  {major = 0, minor = 67, revision = 0, patch = "", version = 180},
+  {major = 0, minor = 68, revision = 0, patch = "", version = 194},
+  {major = 0, minor = 69, revision = 0, patch = "-beta-1", version = 216}
```

Comment:
If we want to reduce the ever growing list here filling up `app.lua`, if there's a good way to stick this table in its own file and then assign it back to release_table I'm open to suggestions.
`version_history.lua`?

Reply:
sure but we can do that any time